.github/workflows/test.yml

```
@@ -271,6 +280,47 @@ jobs:
           command: check
           args: --workspace --all-targets
 
+  build_turborepo:
+    name: Build Turborepo
+    needs: [determine_jobs, rust_prepare]
+    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true' || needs.determine_jobs.outputs.node == 'true'
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Build Turborepo
+        uses: ./.github/actions/build-debug-turborepo
+        with:
+          target: ${{ matrix.os }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: turborepo-${{ matrix.os }}
+          path: |
+            target/debug/turbo
+            target/debug/turbo.exe
+
+  turborepo_integration_tests:
+    name: Cram integration tests
+    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true'
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```

```suggestion
    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true'
    strategy:
      matrix:
        os:
          - name: ubuntu
            runner: ubuntu-latest
          - name: macos
            runner: macos-latest
    runs-on: ${{ matrix.os.runner }}
    name: Cram integration tests on ${{ matrix.os.name }}
```

You could switch to `runner: ubuntu-latest-16-core-oss` if you want this to be faster, but it costs ~$3/h